internals/overlord/state/notices.go

```diff
@@ -270,6 +270,13 @@ func (s *State) AddNotice(userID *uint32, noticeType NoticeType, key string, opt
 	notice.lastData = options.Data
 	notice.repeatAfter = options.RepeatAfter
 
+	// Update the latest warning time cache if needed. There's no need to
+	// actually update atomically here, because the state lock is held.
```

Comment:
Yep, I was about to comment on that. Looks like you've already considered that. If we ever update this without the state mutex, then we'll need to reconsider this.